save-agent/src/nativeMain/kotlin/org/cqfn/save/agent/SaveAgent.kt

```diff
@@ -163,11 +167,14 @@ class SaveAgent(private val config: AgentConfiguration,
     /**
      * Attempt to send execution data to backend, will retry several times, while increasing delay 2 times on each iteration.
      */
-    private suspend fun sendExecutionData(testExecutionDto: TestExecutionDto): Unit = coroutineScope {
+    private suspend fun sendAdditionalDataToBackend(
+        sendExecution: requestExecutionData,
+        testExecutionDto: TestExecutionDto?,
```

Comment:
Why not just pass a function, which produces `HttpResponse`, instead of this complicated logic?

Reply:
Because one of these functions has a parameter

Reply:
Can it be captured when creating a lambda?

Reply:
I suppose, this solution is optimal, at least I don't know how to do this.